src/realm/util/interprocess_mutex.hpp

```diff
@@ -95,8 +95,18 @@ class InterprocessMutex {
     /// InterprocessMutex created on the same file (same inode on POSIX) share the same LockInfo.
     /// LockInfo will be saved in a static map as a weak ptr and use the UniqueID as the key.
     /// Operations on the map need to be protected by s_mutex
-    static std::map<File::UniqueID, std::weak_ptr<LockInfo>> s_info_map;
-    static Mutex s_mutex;
+    static std::map<File::UniqueID, std::weak_ptr<LockInfo>>* s_info_map;
+    static Mutex* s_mutex;
+    /// We control the order of initialization of these static variables,
```

Comment:
Perhaps I'm missing something, but the _order_ these are initialized in doesn't seem significant, only the fact that they're declared and initialized in such a way that their destructor will not run.